compiler/src/iree/compiler/Dialect/Util/Transforms/test/hoist_into_globals.mlir

```diff
@@ -267,6 +267,40 @@ module @hoist_implicit_capture {
 
 // -----
 
+// CHECK-LABEL: @hoist_multi_nested_regions
+module @hoist_multi_nested_regions {
+  // CHECK: util.global private @[[HOISTED_SYM:.*]] : i32
+  // CHECK: util.initializer {
+  // CHECK:       %[[C0:.*]] = arith.constant 0 : i32
+  // CHECK:       %[[CE0:.+]] = "iree_unregistered.const_expr"(%[[C0]])
+  // CHECK:         "iree_unregistered.const_expr"
+  // CHECK:           ^bb0
+  // CHECK:           math.absi
+  // CHECK:       util.global.store %[[CE0]], @[[HOISTED_SYM]] : i32
+  // CHECK:       util.return
+  // CHECK: }
+  // CHECK: util.func public @main
+  util.func public @main() -> (i32) {
+    %0 = arith.constant 0 : i32
+    // CHECK-NOT: arith.constant
+    // CHECK-NOT: iree_unregistered.const_expr
+    // CHECK: %[[VAL:.*]] = util.global.load immutable @[[HOISTED_SYM]] : i32
+    // CHECK: util.return %[[VAL]]
+    %2 = "iree_unregistered.const_expr"(%0) ({
+    ^bb0(%inner0 : i32):
+      %3 = "iree_unregistered.const_expr"(%inner0) ({
+      ^bb0(%inner1 : i32):
+        %4 = math.absi %inner0 : i32
+        "iree_unregistered.yield"(%4) : (i32) -> i32
+      }) : (i32) -> i32
+      "iree_unregistered.yield"(%3) : (i32) -> i32
+    }) : (i32) -> i32
+    util.return %2 : i32
+  }
+}
+
```

Comment:
Can we get a test that looks like
```
    %2 = "iree_unregistered"(%0) ({
    ^bb0(%inner0 : i32):
      %3 = "iree_unregistered.const_expr"(%inner0) ({
      ^bb0(%inner1 : i32):
        %4 = math.absi %inner0 : i32
        "iree_unregistered.yield"(%4) : (i32) -> i32
      }) : (i32) -> i32
      "iree_unregistered.yield"(%3) : (i32) -> i32
    }) : (i32) -> i32
    util.return %2 : i32
```
I'm not sure what we do in this case but I'm guessing we *probably* shouldn't hoist anything here